lib/cc/analyzer/engine_registry.rb

```diff
@@ -3,18 +3,29 @@
 module CC
   module Analyzer
     class EngineRegistry
-      def initialize
+      def initialize(dev_mode = false)
         @path = File.expand_path("../../../../config/engines.yml", __FILE__)
         @config = YAML.safe_load_file(@path)
+        @dev_mode = dev_mode
       end
 
       def [](engine_name)
-        @config[engine_name]
+        if dev_mode?
```

Comment:
Just putting this out there, but I don't think it warrants a change just yet. I'm nervous about this being included in builder because of the potential for arbitrary docker containers to be run. WDYT of overriding this method within builder when we include this to always be false?